File: coverme_test/src/main/java/com/example/coverme_test/diagnostic_screen.java
package com.example.coverme_test;

import android.app.Activity;
import android.content.Context;
import android.graphics.Color;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;


public class diagnostic_screen {

    Context context ;
    grid_draw gr ;

    public diagnostic_screen(Context context){
        this.context = context;
        this.gr = new grid_draw(context);


    }

    public void draw(Activity activity, int numColumns, int numRows ){
        gr.setNumColumns(numColumns);
        gr.setNumRows(numRows);
        gr.setBackgroundColors(Color.GREEN);
        gr.setColor(Color.RED);
        hideSystemUi(activity);
        activity.setContentView(gr);

    }

    private void hideSystemUi(Activity activity){
        activity.getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY|
                 View.SYSTEM_UI_FLAG_LAYOUT_STABLE|
                 View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION|
                 View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN|
                 View.SYSTEM_UI_FLAG_HIDE_NAVIGATION|
                 View.SYSTEM_UI_FLAG_FULLSCREEN);
    }

    public void onTouchItem(MotionEvent event){
        gr.onSelect(event);
    }

    public Double getPorcentage(){
      return   gr.getPorcentage();
    }


}

File: coverme_test/src/main/java/com/example/coverme_test/diagnostic_screen.kt
package com.example.coverme_test

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.graphics.Color
import android.util.Log
import android.view.View
import android.view.Window
import android.view.WindowManager
import com.rm.freedrawview.FreeDrawView
import com.rm.freedrawview.PathDrawnListener

class DiagnosticScreen(context: Context) {

        companion object{


        fun drawGrid(context: Context, activity: Activity, numColumns: Int, numRows: Int, color: Int, fillColor: Int, strokeW: Float){

            var pixelGrid = PixelGridView(context)
            pixelGrid.numColumns = numColumns
            pixelGrid.numRows = numRows
            pixelGrid.fillColor = fillColor
            pixelGrid.backgrounColor = color
            pixelGrid.strokeW = strokeW
            activity.setContentView(pixelGrid)
            hideSystemUI(activity)

        }

        fun hideSystemUI(activity: Activity) {
            activity.window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
                    or View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                    or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                    or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                    or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                    or View.SYSTEM_UI_FLAG_FULLSCREEN)
        }

        fun imprimir() {
            Log.i("Porcentaje", porcenjate.toString())
        }
        }
    }
